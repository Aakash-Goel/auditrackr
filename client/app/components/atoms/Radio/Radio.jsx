import React from 'react';
import { object, arrayOf } from 'prop-types';
import { isEmpty } from 'lodash';

import withStyles from '@material-ui/core/styles/withStyles';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';

import radioStyles from './Radio.style';

/**
 * Type checking - Define prop types
 * @private
 */
const propTypes = {
  options: arrayOf(object).isRequired,
  radioGroupProps: object,
};

/**
 * Define default prop values
 * @private
 */
const defaultProps = {
  radioGroupProps: {},
};

/* eslint-disable react/no-array-index-key */
const CustomRadio = ({ ...props }) => {
  const { radioGroupProps, options } = props;

  if (!isEmpty(radioGroupProps)) {
    return (
      <RadioGroup {...radioGroupProps}>
        {options &&
          options.map((option, index) => (
            <FormControlLabel
              key={index}
              {...option.formControlLabelProps}
              control={<Radio color="primary" {...option.radioProps} />}
            />
          ))}
      </RadioGroup>
    );
  }

  return (
    <>
      {options &&
        options.map((option, index) => (
          <FormControlLabel
            key={index}
            {...option.formControlLabelProps}
            control={<Radio color="primary" {...option.radioProps} />}
          />
        ))}
    </>
  );
};

CustomRadio.propTypes = propTypes;
CustomRadio.defaultProps = defaultProps;

export default withStyles(radioStyles)(CustomRadio);
