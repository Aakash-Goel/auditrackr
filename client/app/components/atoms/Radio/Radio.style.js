/**
 * It seems not required. But kept it, if needed in future.
 *
 * If require to modify the styles, then recommended way is to
 * overrides the mui-theme directly.
 *
 * For stylesheet name and other details, check out:
 *  https://material-ui.com/api/radio/#css
 *
 */
const radioStyles = {};

export default radioStyles;
