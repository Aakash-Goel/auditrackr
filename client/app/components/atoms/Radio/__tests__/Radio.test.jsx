import React from 'react';
import TestRenderer from 'react-test-renderer';

import Radio from '../Radio';

describe('<Radio />', () => {
  it('renders correctly', () => {
    const radioWrapper = TestRenderer.create(
      <Radio
        options={[
          {
            radioProps: {
              checked: true,
              value: 'a',
              name: 'radio-button-demo',
              'aria-label': 'A',
            },
          },
        ]}
      />
    ).toJSON();
    expect(radioWrapper).toMatchSnapshot();
  });

  it('renders correctly if radioGroupProps is passed', () => {
    const radioWrapper = TestRenderer.create(
      <Radio
        radioGroupProps={{
          'aria-label': 'Gender',
          name: 'gender1',
          value: 'female',
        }}
        options={[
          {
            radioProps: {
              checked: false,
              value: 'a',
              name: 'radio-button-demo',
              'aria-label': 'A',
            },
          },
        ]}
      />
    ).toJSON();
    expect(radioWrapper).toMatchSnapshot();
  });
});
