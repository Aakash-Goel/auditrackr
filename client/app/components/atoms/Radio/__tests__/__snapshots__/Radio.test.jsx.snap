// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> renders correctly 1`] = `
<label
  className="MuiFormControlLabel-root-2"
>
  <span
    aria-label="A"
    className="MuiButtonBase-root-23 MuiIconButton-root-17 MuiPrivateSwitchBase-root-13 MuiRadio-root-8 MuiRadio-colorPrimary-11 MuiPrivateSwitchBase-checked-14 MuiRadio-checked-9"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label-22"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-26"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <input
        checked="a"
        className="MuiPrivateSwitchBase-input-16"
        name="radio-button-demo"
        onChange={[Function]}
        type="radio"
        value="a"
      />
    </span>
    <span
      className="MuiTouchRipple-root-71"
    />
  </span>
  <span
    className="MuiTypography-root-35 MuiTypography-body1-44 MuiFormControlLabel-label-7"
  />
</label>
`;

exports[`<Radio /> renders correctly if radioGroupProps is passed 1`] = `
<div
  aria-label="Gender"
  className="MuiFormGroup-root-78"
  role="radiogroup"
>
  <label
    className="MuiFormControlLabel-root-2"
  >
    <span
      aria-label="A"
      className="MuiButtonBase-root-23 MuiIconButton-root-17 MuiPrivateSwitchBase-root-13 MuiRadio-root-8 MuiRadio-colorPrimary-11 MuiPrivateSwitchBase-checked-14 MuiRadio-checked-9"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label-22"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-26"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <input
          checked="a"
          className="MuiPrivateSwitchBase-input-16"
          name="radio-button-demo"
          onChange={[Function]}
          type="radio"
          value="a"
        />
      </span>
      <span
        className="MuiTouchRipple-root-71"
      />
    </span>
    <span
      className="MuiTypography-root-35 MuiTypography-body1-44 MuiFormControlLabel-label-7"
    />
  </label>
</div>
`;
